### PR TITLE
Add OpenNebulaEndpoint

### PR DESCRIPTION
**Pull request by rkuchumov**
OpenNebulaEndpoint is based on existing OcciEndpoint. VM creation
(createInstances) and termination (stopVM) are now working, other features
(validating endpoint URL, accessing VMs metadata and interfaces) are
not implemented and return error.  Some parameters, such as HostCert,
HostKey or TemplateID, are hardcoded.

### CODE REVIEW

File: VMDIRAC/Resources/Cloud/OpenNebulaEndpoint.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
+###########################################################
+# $HeadURL$
+# File: OcciEndpoint.py
+# Author: A.T.
+###########################################################
+
+"""
+   OcciEndpoint is Endpoint base class implementation for the Occi cloud service.
+"""
+
```
Comment (fstagni) [inline VMDIRAC/Resources/Cloud/OpenNebulaEndpoint.py]: This is not for OCCI :)

and the line  # $HeadURL$ is not needed.
... and you are not A.T. :)

File: VMDIRAC/Resources/Cloud/OpenNebulaEndpoint.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+__RCSID__ = 'e069e3a (2018-05-15 22:19:34 +0200) Andrei Tsaregorodtsev <atsareg@in2p3.fr>'
```
Comment (fstagni) [inline VMDIRAC/Resources/Cloud/OpenNebulaEndpoint.py]: This should be __RCSID__ = "$Id$"

File: VMDIRAC/Resources/Cloud/OpenNebulaEndpoint.py [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+class OpenNebulaEndpoint( Endpoint ):
+  """ OCCI implementation of the Cloud Endpoint interface
+  """
+
+  def __init__( self, parameters = {} ):
+    """
+    """
+    super( OpenNebulaEndpoint, self ).__init__(parameters = parameters, bootstrapParameters=parameters)
+
+    print "AT >>> parameters", parameters
```
Comment (fstagni) [inline VMDIRAC/Resources/Cloud/OpenNebulaEndpoint.py]: Remove this line.

File: VMDIRAC/Resources/Cloud/OpenNebulaEndpoint.py [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+        openstack node id ( not uuid ! )
+
+    :return: S_OK( status ) | S_ERROR
+    """
+    return S_ERROR('getVMStatus is not implemented')
+
+  def getVMNetworks( self, networkNames = [] ):
+    """ Get a network object corresponding to the networkName
+
+    :param str networkName: network name
```
Comment (fstagni) [inline VMDIRAC/Resources/Cloud/OpenNebulaEndpoint.py:205]: `networkNames`

File: VMDIRAC/Resources/Cloud/OpenNebulaEndpoint.py [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+    """
+    Get the status for a given node ID. libcloud translates the status into a digit
+    from 0 to 4 using a many-to-one relation ( ACTIVE and RUNNING -> 0 ), which
+    means we cannot undo that translation. It uses an intermediate states mapping
+    dictionary, SITEMAP, which we use here inverted to return the status as a
+    meaningful string. The five possible states are ( ordered from 0 to 4 ):
+    RUNNING, REBOOTING, TERMINATED, PENDING & UNKNOWN.
+
+    :Parameters:
+      **uniqueId** - `string`
```
Comment (fstagni) [inline VMDIRAC/Resources/Cloud/OpenNebulaEndpoint.py:195]: No, it's `nodeID`

File: VMDIRAC/Resources/Cloud/OpenNebulaEndpoint.py [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+    if ok:
+      return S_OK((payload,None))
+
+    return S_ERROR('one.template.instantiate failed with code: {} message: {} ', error, payload)
+
+
+  def getVMIDs( self ):
+    """ Get all the VM IDs on the endpoint
+
+    :return: list of VM ids
```
Comment (fstagni) [inline VMDIRAC/Resources/Cloud/OpenNebulaEndpoint.py:181]: It's not what it does, it seems to me.

File: VMDIRAC/Resources/Cloud/OpenNebulaEndpoint.py [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+    """
+    return S_ERROR('getVMNetworkInterface is not implemented')
+
+  def stopVM( self, nodeID, publicIP = '' ):
+    """
+    Given the node ID it gets the node details, which are used to destroy the
+    node making use of the libcloud.openstack driver. If three is any public IP
+    ( floating IP ) assigned, frees it as well.
+
+    :Parameters:
```
Comment (fstagni) [inline VMDIRAC/Resources/Cloud/OpenNebulaEndpoint.py]: These are not the parameters' names. This is true for all the docstrings.

**Comment by rkuchumov** [comment]
@fstagni Ok, I've fixed some of the issues. I don't see any point in modifying docs for stubbed methods, I agree only to remove them. Your comments should apply to `VMDIRAC/Resources/Cloud/OcciEndPoint.py` file as it was originally copy-pasted  to this file.

**Comment by IgorPelevanyuk** [comment]
@fstagni It is just a first step. Maybe it could be merged in some temporary branch? After that, I will add configuration use and refine it.

**Comment by atsareg** [comment]
Replaced by  #112

**Pull request closed without merging**